Data-Structures/Linked-List/MiddleOfLinkedList.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * A LinkedList based solution for finding middle node of linked list.
+ * https://afteracademy.com/blog/middle-of-the-linked-list
+ */
+class MiddleOfLL {
+  solution (head) {
+  /*
+    Problem Statement:
+    Given the head of a singly linked list, return the middle node of the linked list.
+    If there are two middle nodes, return the second middle node.
+
+    Link for the Problem: https://leetcode.com/problems/middle-of-the-linked-list/
+    */
+    let fast = head
+    let slow = head
+
+    if (head.next == null) { return head }
```

Comment:
This line is superfluous: If `head.next == null`, `fast.next` will be `null` in the first iteration, causing the `while` loop to not be entered at all. `slow`, which is returned, then is `head`.

Reply:
Yeah, Right! Makes sense.